test/UnitTest/Microsoft.AspNet.OData.Test.Shared/TestCommon/Models/TestDataModels.cs

```diff
@@ -54,4 +57,26 @@ public class AutoExpandChoiceOrder
         public int Id { get; set; }
         public string Name { get; set; }
     }
+
+    public class NewCustomerUnmapped
+    {
+        [Key]
+        public int Id { get; set; }
+        [IgnoreDataMember]
+        public string Name { get; set; }
+        [NotMapped]
+        public int Age { get; set; }
+    }
+
+    [DataContract]
+    public class NewCustomerDataContract
+    {
+        [Key]
+        public int Id { get; set; }
+        [IgnoreDataMember]
+        [DataMember]
+        public string Name { get; set; }
```

Comment:
Just to cover all of the possibilities, should probably have a property with just the [DataMember] attribute and a property with no attributes.